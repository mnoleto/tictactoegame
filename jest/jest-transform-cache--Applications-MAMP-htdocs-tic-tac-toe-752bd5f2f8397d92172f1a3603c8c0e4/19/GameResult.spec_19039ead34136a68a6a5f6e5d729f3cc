/* istanbul ignore next */"use strict";GameResult.spec.js;