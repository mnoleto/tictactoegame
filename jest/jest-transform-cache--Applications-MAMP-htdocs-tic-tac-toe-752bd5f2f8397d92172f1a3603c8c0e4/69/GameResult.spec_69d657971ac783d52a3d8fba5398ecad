/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_sinon = require('sinon'); /* istanbul ignore next */var _sinon2 = _interopRequireDefault(_sinon);
var /* istanbul ignore next */_GameResult = require('./GameResult'); /* istanbul ignore next */var _GameResult2 = _interopRequireDefault(_GameResult);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// GameResult is responsible for render the game grid
describe('GameResult', function () {
  var props = /* istanbul ignore next */void 0,mountedGameResult = /* istanbul ignore next */void 0;
  var gameResult = function gameResult() {
    if (!mountedGameResult) {
      mountedGameResult = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_GameResult2.default, props));

    }
    return mountedGameResult;
  };

  beforeEach(function () {
    props = {
      onNewGameClick: jest.fn(),
      onNewRoundClick: jest.fn(),
      players: ['Player 1', 'Player 2'],
      result: { status: 'finished' } };

    mountedGameResult = undefined;
  });

  it('render the GameResult', function () {
    var gameResultWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_GameResult2.default, props));
    expect(gameResultWrapper).toMatchSnapshot();
  });

  it('div contains everything else that gets rendered', function () {
    var gameResultElement = gameResult().find('div');
    var wrappingDiv = gameResultElement.first();
    expect(wrappingDiv.hasClass('game-result')).toBeTruthy();
    expect(wrappingDiv.children()).toEqual(gameResult().children());
  });

  it('should have a p element with a message to the winner', function () {

  });

  it('should have a p element with a draw message', function () {

  });

  describe('buttons', function () {
    it('should have a button wrapper', function () {
      var buttonWrapperElement = gameResult().find('div.buttons');
      expect(buttonWrapperElement.length).toEqual(1);
    });

    it('should have two buttons', function () {
      var buttonElements = gameResult().find('button');
      expect(buttonElements.length).toEqual(2);
    });

    it('simulate new round click', function () {
      props.onNewGameClick = /* istanbul ignore next */_sinon2.default.spy();
      var buttonWrapperElement = gameResult().find('div.buttons');
      var newRoundButton = buttonWrapperElement.childAt(0);

      newRoundButton.simulate('click');
      expect(props.onNewGameClick.callCount).toEqual(1);
    });

    it('simulate new game click', function () {
      props.onNewGameClick = /* istanbul ignore next */_sinon2.default.spy();
      var buttonWrapperElement = gameResult().find('div.buttons');
      var newGameButton = buttonWrapperElement.childAt(1);

      newGameButton.simulate('click');
      expect(props.onNewGameClick.callCount).toEqual(1);
    });
  });

});