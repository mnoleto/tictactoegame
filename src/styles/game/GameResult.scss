@import '../_colors.scss';

.game-result {
  margin-bottom: 50px;
  p {
    font-size: 500%;
    text-align: center;
  }
}

.new-game-button {
  background-color: lighten($blue, 20%);
  border: 0px;
  color: $dark-blue;
  display: block;
  font-size: 100%;
  margin: 15px auto 10px;
  padding: 10px 20px;
}