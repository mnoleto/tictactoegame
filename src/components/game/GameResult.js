import React, { Component, PropTypes } from 'react';
import '../../styles/game/GameResult.scss';

// GameResult is responsible for render the game result
class GameResult extends Component {
  constructor(props) {
    super(props);
    this.renderMessage = this.renderMessage.bind(this);
  }

  renderMessage() {
    const { players, result } = this.props;
    if(result.status === 'draw') {
      const draw = ['It`s a draw!'];
      return draw.map((value, index) => <span key={'message_' + index}>{value}</span>);
    } else {
      const winnerName = (!result.draw && result.winner === 'X') ? players[0] : players[1];
      const winner = ['The winner is ', <strong>{winnerName}</strong>, '.'];
      return winner.map((value, index) => <span key={'message_' + index}>{value}</span>);
    }
  }

  render() {
    const { onNewGameClick, onNewRoundClick } = this.props;
    
    return (
      <div className="game-result">
        <p>{this.renderMessage()}</p>
        <div className="buttons">
          <button type="button" className="new-round-button" onClick={onNewRoundClick}>New Round</button>
          <button type="button" className="new-game-button" onClick={onNewGameClick}>New Game</button>
        </div>
      </div>
    );
  }
}

GameResult.propTypes = {
  onNewGameClick: PropTypes.func.isRequired,
  onNewRoundClick: PropTypes.func.isRequired,
  players: PropTypes.array.isRequired,
  result: PropTypes.object.isRequired
}

export default GameResult;