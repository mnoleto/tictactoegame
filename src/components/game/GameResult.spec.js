import React from 'react';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';
import GameResult from './GameResult';

// GameResult is responsible for render the game result
describe('<GameResult />', () => {
  let props, mountedGameResult;
  const gameResult = () => {
    if(!mountedGameResult) {
      mountedGameResult = mount(
        <GameResult {...props} />
      );
    }
    return mountedGameResult;
  };

  beforeEach(() => {
    props = {
      onNewGameClick: jest.fn(),
      onNewRoundClick: jest.fn(),
      players: ['Player 1', 'Player 2'],
      result: {status: 'finished', winner: 'O'}
    };
    mountedGameResult = undefined;
  });

  it('render the GameResult', () => {
    const gameResultWrapper = shallow(<GameResult {...props} />);
    expect(gameResultWrapper).toMatchSnapshot();
  });

  it('div contains everything else that gets rendered', () => {
    const gameResultElement = gameResult().find('div');
    const wrappingDiv = gameResultElement.first();
    expect(wrappingDiv.hasClass('game-result')).toBeTruthy();
    expect(wrappingDiv.children()).toEqual(gameResult().children());
  });

  it('should have a p element with a message to the winner', () => {
    const messageElement = gameResult().find('p');
    expect(messageElement.length).toEqual(1);
    expect(messageElement.text()).toEqual('The winner is Player 2.');
  });

  it('should have a p element with a draw message', () => {
    props.result = {status: 'draw'};
    const messageElement = gameResult().find('p');
    expect(messageElement.length).toEqual(1);
    expect(messageElement.text()).toEqual('It`s a draw!');
  });

  describe('buttons', () => {
    it('should have a button wrapper', () => {
      const buttonWrapperElement = gameResult().find('div.buttons');
      expect(buttonWrapperElement.length).toEqual(1);
    });

    it('should have two buttons', () => {
      const buttonElements = gameResult().find('button');
      expect(buttonElements.length).toEqual(2);
    });

    it('simulate new round click', () => {
      props.onNewRoundClick = sinon.spy();
      const buttonWrapperElement = gameResult().find('div.buttons');
      const newRoundButton = buttonWrapperElement.childAt(0);

      newRoundButton.simulate('click');
      expect(props.onNewRoundClick.callCount).toEqual(1);
    });

    it('simulate new game click', () => {
      props.onNewGameClick = sinon.spy();
      const buttonWrapperElement = gameResult().find('div.buttons');
      const newGameButton = buttonWrapperElement.childAt(1);

      newGameButton.simulate('click');
      expect(props.onNewGameClick.callCount).toEqual(1);
    });
  });
});