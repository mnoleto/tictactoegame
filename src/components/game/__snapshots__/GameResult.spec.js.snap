exports[`<GameResult /> render the GameResult 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="game-result">
    <p>
        <span>
            The winner is 
        </span>
        <span>
            <strong>
                Player 2
            </strong>
        </span>
        <span>
            .
        </span>
    </p>
    <div
        className="buttons">
        <button
            className="new-round-button"
            onClick={[Function]}
            type="button">
            New Round
        </button>
        <button
            className="new-game-button"
            onClick={[Function]}
            type="button">
            New Game
        </button>
    </div>
</div>,
  "nodes": Array [
    <div
      className="game-result">
      <p>
            <span>
                  The winner is 
            </span>
            <span>
                  <strong>
                        Player 2
                  </strong>
            </span>
            <span>
                  .
            </span>
      </p>
      <div
            className="buttons">
            <button
                  className="new-round-button"
                  onClick={[Function]}
                  type="button">
                  New Round
            </button>
            <button
                  className="new-game-button"
                  onClick={[Function]}
                  type="button">
                  New Game
            </button>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <GameResult
        onNewGameClick={[Function]}
        onNewRoundClick={[Function]}
        players={
                Array [
                        "Player 1",
                        "Player 2",
                      ]
        }
        result={
                Object {
                        "status": "finished",
                        "winner": "O",
                      }
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": GameResult {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onNewGameClick": [Function],
          "onNewRoundClick": [Function],
          "players": Array [
            "Player 1",
            "Player 2",
          ],
          "result": Object {
            "status": "finished",
            "winner": "O",
          },
        },
        "refs": Object {},
        "renderMessage": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="game-result">
          <p>
                    <span>
                              The winner is 
                    </span>
                    <span>
                              <strong>
                                        Player 2
                              </strong>
                    </span>
                    <span>
                              .
                    </span>
          </p>
          <div
                    className="buttons">
                    <button
                              className="new-round-button"
                              onClick={[Function]}
                              type="button">
                              New Round
                    </button>
                    <button
                              className="new-game-button"
                              onClick={[Function]}
                              type="button">
                              New Game
                    </button>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="game-result">
          <p>
                    <span>
                              The winner is 
                    </span>
                    <span>
                              <strong>
                                        Player 2
                              </strong>
                    </span>
                    <span>
                              .
                    </span>
          </p>
          <div
                    className="buttons">
                    <button
                              className="new-round-button"
                              onClick={[Function]}
                              type="button">
                              New Round
                    </button>
                    <button
                              className="new-game-button"
                              onClick={[Function]}
                              type="button">
                              New Game
                    </button>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <GameResult
    onNewGameClick={[Function]}
    onNewRoundClick={[Function]}
    players={
        Array [
            "Player 1",
            "Player 2",
          ]
    }
    result={
        Object {
            "status": "finished",
            "winner": "O",
          }
    } />,
}
`;
